[18.0][MIG] fieldservice_sale_agreement_equipment_stock: Migration to 18.0

vvrossem opened the pull request:
Depends on:

* https://github.com/OCA/agreement/pull/51

ivantodorovich committed:
[ADD] fieldservice_sale_agreement_equipment_stock

mymage committed:
Added translation using Weblate (Italian)

vvrossem committed:
[MIG] fieldservice_sale_agreement_equipment_stock: Migration to 18.0

ivantodorovich commented on fieldservice_sale_agreement_equipment_stock/tests/test_fsm_sale_agreement_equipment_stock.py:
```diff
@@ -17,7 +17,7 @@ def setUpClass(cls):
             {
                 "name": "FSM Product",
-                "type": "product",
+                "type": "consu",
```
I'm not sure it makes a difference, but to keep the exact same previous behavior, set also `is_storable`: True

ivantodorovich reviewed (changes requested):
CI is red, could you take a look? 🙏🏻

vvrossem commented on fieldservice_sale_agreement_equipment_stock/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+    "category": "Field Service",
+    "depends": [
+        "agreement_sale",
+        "fieldservice_sale",
```
@ivantodorovich shouldn't `fieldservice_agreement` be listed too since `fsm.equipment.agreement_id` is defined in that module?

ivantodorovich replied:
Yes, I think you're right!

vvrossem committed:
[FIX] fieldservice_sale_agreement_equipment_stock: AccessError

Fixes
```
odoo.exceptions.AccessError: You are not allowed to access 'Field Service Order' (fsm.order) records.

This operation is allowed for the following groups:
        - Field Service/Dispatcher
        - Field Service/User (only own documents)

Contact your administrator to request access if necessary.
```

vvrossem committed:
[FIX] fieldservice_sale_agreement_equipment_stock: ValueError

Fixes `ValueError: Invalid field 'agreement_id' on model 'fsm.equipment'`

ivantodorovich commented:
Depends on:
- https://github.com/OCA/agreement/pull/51

BhaveshHeliconia reviewed (approved):
Functional review LGTM!

vvrossem commented:
:information_source: @ivantodorovich I've rebased now that `agreement_sale` is merged

ivantodorovich commented:
/ocabot migration fieldservice_sale_agreement_equipment_stock
/ocabot merge nobump
